prow/config/config.go

```diff
@@ -434,52 +434,27 @@ func Load(prowConfig, jobConfig string) (c *Config, err error) {
 	return c, nil
 }
 
-// loadConfig loads one or multiple config files and returns a config object.
-func loadConfig(prowConfig, jobConfig string) (*Config, error) {
-	stat, err := os.Stat(prowConfig)
-	if err != nil {
-		return nil, err
-	}
-
-	if stat.IsDir() {
-		return nil, fmt.Errorf("prowConfig cannot be a dir - %s", prowConfig)
-	}
-
-	var nc Config
-	if err := yamlToConfig(prowConfig, &nc); err != nil {
-		return nil, err
-	}
-	if err := parseProwConfig(&nc); err != nil {
-		return nil, err
-	}
-
-	// TODO(krzyzacy): temporary allow empty jobconfig
-	//                 also temporary allow job config in prow config
-	if jobConfig == "" {
-		return &nc, nil
-	}
-
-	stat, err = os.Stat(jobConfig)
+// ReadJobConfig reads the JobConfig yaml, but does not expand or validate it.
```

Comment:
You want this since you don't care to start the config agent loader? Seems like you could have gotten away with just starting it and reading once, but SGTM

Reply:
If the config agent will give me the config without expansions (I didn't check) I am probably happy to use that instead — my issue is that I'm using what I read in to write it back out later. If expansion happens, then the result is a) hard to read, and also b) actually fails validation (because the presets are still attached and cause duplication errors when re-expanded).

I believe you have used this tooling to parse/write back your files, which suggests I've missed something.

Reply:
It won't, there's a bunch of computed stuff from the config agent loading. The generated security jobs had a lot of problems with this

Reply:
We don't use presets, so we side-stepped the issue.